0000-catalog.md

```diff
@@ -0,0 +1,779 @@
+    CatalogProductCreateAction catalog_product_create = 6;
+    CatalogProductUpdateAction catalog_product_update = 7;
+    CatalogProductDeleteAction catalog_product_delete = 8;
+    CatalogProductSetStatusAction set_catalog_product_status = 9;
+}
+```
+
+## Catalog Actions
+
+### CatalogCreateAction
+
+CatalogCreateAction adds a new catalog to state. The transaction should be
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the create transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+
+```
+message CatalogCreateAction {
+    // GS1 Company Prefix from owner + catalog_id are use as
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id already exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_create_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogCreateAction must include:
+
+Address of the Agent submitting the transaction
+Address of the Organization the Catalog is being created for
+Address of the Catalog to be created
+
+The outputs for CatalogCreateAction must include:
+
+Address of the Catalog created
+
+### CatalogUpdateAction
+
+CatalogUpdateAction adds a new catalog to state. The transaction should be
+submitted by an agent, which is identified by its signing key, acting on behalf 
+of the organization that corresponds to the owner in the update transaction. 
+(Organizations and agents are defined by the Pike smart contract.)
+
+```
+message CatalogUpdateAction {
+    // GS1 Company Prefix from owner + catalog_id are use as
+    // a composite key for determining the state address
+    string owner = 1;
+    string catalog_id = 2;
+    string name = 3;
+    repeated PropertyValues properties = 4;
+}
+```
+
+Validation requirements:
+
+- If a catalog with catalog_id exists the transaction is invalid.
+- The signer of the transaction must be an agent in the Pike state and must
+  belong to an organization in Pike state, otherwise the transaction is invalid.
+- The agent must have the permission can_update_catalog for the organization,
+  otherwise the transaction is invalid.
+
+The inputs for CatalogUpdateAction must include:
+
+Address of the Agent submitting the transaction
+Address of the Organization the Catalog is being created for
+Address of the Catalog to be updated
+
+The outputs for CatalogUpdateAction must include:
+
+Address of the Catalog to be updated
+
+### CatalogDeleteAction
+
+CatalogDeleteAction adds a new catalog to state. The transaction should be
```

Comment:
deletes a catalog

Reply:
✅

Reply:
It now says "deletes a new catalog to state" should say "deletes an existing catalog to state"

Reply:
added ✅